src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CodeFixes/HttpContextRefactor/HttpContextRefactorCodeFixProvider.cs

```diff
@@ -0,0 +1,297 @@
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CodeFixes
+{
+    public abstract class HttpContextRefactorCodeFixProvider<TInvocationNode, TArgument> : CodeFixProvider
+        where TInvocationNode : SyntaxNode
+        where TArgument : SyntaxNode
+    {
+        private const string DefaultArgumentName = "currentContext";
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds
+        {
+            get { return ImmutableArray.Create(HttpContextCurrentAnalyzer.DiagnosticId); }
+        }
+
+        public sealed override FixAllProvider GetFixAllProvider()
```

Comment:
should we just make a new base class we reuse? I think we're putting this in every codefixer